packages/invoices/src/OutboundShipment/DetailView/DetailView.tsx

```diff
@@ -1,145 +1,43 @@
-    dispatch(OutboundAction.onSortBy(column));
-  };
-
-  const onFlattenRows = () => {
-    dispatch(OutboundAction.flattenRows());
-  };
-
-  const onGroupRows = () => {
-    dispatch(OutboundAction.groupRows());
-  };
-
-  return {
-    draft,
-    save,
-    dispatch,
-    onChangeSortBy,
-    sortBy: state.sortBy,
-    onFlattenRows,
-    onGroupRows,
-  };
-};
-
 export const DetailView: FC = () => {
-  const { draft } = useDraftOutbound();
-
-  const { prefetchListByName } = useItemsList({
-    initialSortBy: { key: 'name' },
-  });
-
-  const [selectedItem, setSelectedItem] = React.useState<{
-    item: OutboundShipmentSummaryItem | null;
-    editing: boolean;
-  }>({ item: null, editing: false });
-
-  const itemModalControl = useToggle();
-
-  const findNextItem = (currentItem: OutboundShipmentSummaryItem | null) => {
-    if (!currentItem) return null;
-    const currentItemIdx = draft.items.findIndex(
-      item => item.id === currentItem?.id
-    );
-
-    return draft.items[(currentItemIdx + 1) % draft.items.length];
-  };
-
-  const onNext = () => {
-    if (selectedItem.editing) {
-      const nextItem = findNextItem(selectedItem?.item);
-
-      if (nextItem) {
-        setSelectedItem({ item: nextItem, editing: true });
-
-        const toPrefetch = findNextItem(nextItem);
-        if (toPrefetch) prefetchListByName(toPrefetch.itemName);
-      }
-    } else {
-      setSelectedItem({ item: null, editing: false });
-    }
-
-    return true;
-  };
-
-  const onChangeSelectedItem = (newItem: Item | null) => {
-    if (!newItem) return setSelectedItem({ item: null, editing: false });
-
-    // Try and find the outbound summary row that matches the new item
-    const item = draft.items.find(
-      summaryItem => summaryItem.itemId === newItem.id
-    );
-
-    // If we found it, set the selected item.
-    if (item) {
-      setSelectedItem({ item, editing: true });
-    } else {
-      // otherwise, set the selected item to a newly created summary row.
-      setSelectedItem({ item: itemToSummaryItem(newItem), editing: false });
-    }
-  };
+  const { entity, mode, onOpen, onClose, isOpen } = useEditModal<Item>();
 
   const onRowClick = (item: InvoiceLine | InvoiceItem) => {
-    console.log(item);
+    onOpen(toItem(item));
   };
 
-  return draft ? (
+  return (
     <TableProvider createStore={createTableStore}>
-      <AppBarButtons onAddItem={itemModalControl.toggleOn} />
+      <AppBarButtons onAddItem={() => onOpen()} />
 
-      <ItemDetailsModal
-        draft={draft}
-        isEditMode={selectedItem.editing}
-        onNext={onNext}
-        summaryItem={selectedItem?.item}
-        isOpen={itemModalControl.isOn}
-        onClose={itemModalControl.toggleOff}
-        onChangeItem={onChangeSelectedItem}
-        upsertInvoiceLine={line => draft.upsertLine?.(line)}
-        isOnlyItem={draft.items.length === 1}
-      />
+      {isOpen && (
```

Comment:
I couldn't see why this was necessary - isn't the `isOpen` prop enough?

Reply:
The intention was to ensuring the modal is unmounted, which will cause any local state to reset without explicitly resetting it all. Obviously the implictness of it is not very clear 🤔